middleware.go

```diff
@@ -197,6 +197,7 @@ func (r *oauthProxy) authenticationMiddleware() func(http.Handler) http.Handler
 					&oidc3.Config{
 						ClientID:          r.config.ClientID,
 						SkipClientIDCheck: true,
+						SkipIssuerCheck:   true,
```

Comment:
I'm concerned about existing users that rely on this issuer check for security. Perhaps we can add a config flag that allows `SkipIssuerCheck`, but default it to `false`? Or maybe we need to discuss how we want to approach backward compatibility

Reply:
yup, we can do that altough this checking is in fact bug, we can provide flags to have users more fluent migration

Reply:
i added flags